src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -411,7 +411,8 @@ export class PtyService extends Disposable implements IPtyService {
 			fixedDimensions: persistentProcess.fixedDimensions,
 			environmentVariableCollections: persistentProcess.processLaunchOptions.options.environmentVariableCollections,
 			reconnectionOwner: persistentProcess.shellLaunchConfig.reconnectionOwner,
-			task: persistentProcess.shellLaunchConfig.task
+			task: persistentProcess.shellLaunchConfig.task,
+			waitOnExit: persistentProcess.shellLaunchConfig.waitOnExit
```

Comment:
I don't think we're persisting `hideFromUser` and `isFeatureTerminal` yet.

Reply:
also not persisting `Type` though was thinking we could just use `reconnectionOwner` as the description if there is one